### ISSUE TITLE
Feature Requests: Rewind phases

### ISSUE THREAD

**Issue by benlivinlife** [Card-Forge/forge#4600]
I think it would be really nice and make this awesome app a little better and a little more sandboxy, especially for testing decks, to add a true back/undo button when going through phases of a turn. Ive had a lot of issues, especially with learning the ins and outs, with skipping past my second main phase in particular or accidently tapping for mana, thinking I could play something that I couldnt and then going to a different phase only to have all of my tapped mana go away.

**Comment by TheLastNarwhal** [comment]
When you tap for mana you can undo it by simply clicking the "undo" button next to the "okay" button. Some exceptions to this are of course where costs are paid to get that mana and/or life is lost to get that mana.

As for the "rollback button," Forge is currently incapable of supporting a feature like that due to how it's coded. Or at least I believe that's what  the dev team has said about this in the past.

**Comment by benlivinlife** [comment]
Ah ok. unfortunately I do know you can untap lands by hitting undo, but thats only if you are still in the same phase of a turn. for example, sometimes I will tap lands to play a spell thinking Im in my 2nd main only to realize Im still in the post combat phase and then forget to untap them before hitting ok to move to the 2nd main and lose all my mana. So its moslty a me issue and I will just have to slow down and be more careful. This is also my first week using this tool so I am sure with more use it wont be as much of a problem for me. Just thought it would be a nice addition if possible.

**Comment by TheLastNarwhal** [comment]
Yeah, slowing down does help a lot, especially when first starting out. There is setting to alert you to unused mana being emptied from your mana pool. I would strongly recommend you use this as it does come in handy. It can be found in Preferences and under Gameplay.

![Screenshot_20240126_140405](https://github.com/Card-Forge/forge/assets/92389638/9a989b21-4f53-4733-8aaf-b14075b2f330)

**Comment by benlivinlife** [comment]
oh Awesome! I will definitely turn that on then. Thank you!

**Comment by Nymra** [comment]
yeah, this is a very pressing issue. 

its PvE and the game should be fun, not torture... going back a step should be a top priority... especially since its not possible to customize the game further to make it more convenient to play (the empty mana pool warning is a nice option, but it should also exist a "warn on board wipe" and other stuff. 
all that would not be necessary if we could just go back a step.

**Comment by MorihiroS** [comment]
@Hanmac, need keep label.

Although I sincerely doubt that requests will be handled here at all anymore

**Comment by Hanmac** [comment]
> @Hanmac, need keep label.
> 
> Although I sincerely doubt that requests will be handled here at all anymore

New Sets/Release have more Priority

**Comment by stevekm** [comment]
I found the "Undo" button in the "Game" menu but its pretty much useless because it does not let you go back to a previous phase. Repeatedly now I have accidentally skipped my Declare Attackers phase because I accidentally clicked something in the UI. I would consider this a serious bug in the game because it makes no sense for the game to be taking actions that you did not tell it to (e.g. skip/end whatever phase you are in) and then also not allow you to go back and correct the mistake.

<img width="267" height="161" alt="Image" src="https://github.com/user-attachments/assets/cccc3986-bb6b-452c-a409-521d500b2e80" />

**Comment by CTimmerman** [comment]
Undoing phases could also be useful for simply guessing and evaluating effects instead of coding special rules like #8347.

**Comment by GeoDeWitt** [comment]
> Undoing phases could also be useful for simply guessing and evaluating effects instead of coding special rules like [#8347](https://github.com/Card-Forge/forge/pull/8347).

This would be amazing! Especially when I am clicking too fast in infinite loops and accidentally click the wrong thing.
